GVFS/GVFS.Common/Git/GitProcess.cs

```diff
@@ -461,9 +461,9 @@ public Result CreateBranchWithUpstream(string branchToCreate, string upstreamBra
             return this.InvokeGitAgainstDotGitFolder("branch " + branchToCreate + " --track " + upstreamBranch);
         }
 
-        public Result ForceCheckout(string target)
+        public Result ForceCheckout(string target, string paths = null)
         {
-            return this.InvokeGitInWorkingDirectoryRoot("checkout -f " + target, useReadObjectHook: false);
+            return this.InvokeGitInWorkingDirectoryRoot($"checkout -f {target} {paths}", useReadObjectHook: false);

```

Comment:
I had to check to make sure that `null` would write an empty string here, and it does. Carry on.